src/v/cluster/partition_balancer_backend.cc

```diff
@@ -247,12 +247,24 @@ void partition_balancer_backend::on_topic_table_update() {
 void partition_balancer_backend::on_health_monitor_update(
   const node_health_report& report,
   std::optional<ss::lw_shared_ptr<const node_health_report>> old_report) {
+    auto has_log_data_size = [](const node_health_report& r) {
+        return r.local_state.log_data_size.has_value();
+    };
+
     if (!old_report) {
         vlog(
           clusterlog.debug,
           "health report for node {} appeared, scheduling tick",
           report.id);
 
+        maybe_rearm_timer(/*now=*/true);
+    } else if (
+      !has_log_data_size(*old_report.value()) && has_log_data_size(report)) {
+        vlog(
+          clusterlog.debug,
+          "log data size for node {} appeared, scheduling tick",
+          report.id);
+
         maybe_rearm_timer(/*now=*/true);
     }
```

Comment:
Why do we need to rearm it in both situations? Can't we only rearm it only if `has_log_data_size(report) && !(old_report && has_log_data_size(*old_report.value()))`?

Reply:
In theory space management can be disabled, then we need to rearm even if `!has_log_data_size(report)`. Shouldn't be a big problem as exiting is cheap if we exit early due to insufficient disk info.